### DISCUSSION TITLE
Trouble implementing orm relationship

### DISCUSSION THREAD

**Discussion by sean-janus** [sqlalchemy/sqlalchemy#6732] [Usage Questions] [answered]
I am unable to figure out what I am doing wrong in implementing orm relationships. I am using the following versions:
Python 3.8
SQLAlchemy==1.4.14
MS SQL Server / SQLEXPRESS 15.0.280

I posted the details of my code here in stackoverflow, and I can reproduce it here.
https://stackoverflow.com/questions/68292548/sqlalchemy-orm-relationship-stopped-working

Please help point me in the right direction. I was able to get some relationships working earlier... but now I am unable to reproduce my prior success. I have also been modifying the models/schemas and underlying tables, so perhaps this is an issue with the model(s) being cached??

Thank you in advance,
Sean

**Comment by nuvic** [comment]
I implemented your code with just sqlalchemy, and did get the relationships working. This query `db.query(models.TestP).all()` returns a list of `TestP` objects as expected. Each `TestP` object also has an associated `Test` object. If you run it in a debugger (in crud.py) and do something like this...
```python
res = db.query(models.TestP).all()
for r in res:
  print(r)
  print(r.name)
  print(r.pk_id)
  print(r.fk_id)
  print(r.test.test_id) # Test object
  print(r.test.test_name) # Test object
```
You should be able to see that the query does return the expected relationships. Since it's not returning the correct json at your url, it could be pydantic's serialization that's not working as expected.

**Comment by CaselIT** [comment, marked as answer]
A couple of things:

```py
def test_rel(db: Session) -> List[schemas.TestP]:
    return db.query(models.TestP).all()
```

This returns a `List[model.TestP]`

Also the relationships are lazy loaded by default. Look at https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html for the eager loading options

**Reply by sean-janus** [reply]
Thank you! This was exactly what I was looking for. I was unaware that relationships were lazy loaded by default. My code is using  pydantic to validate the response and my queries were failing validation because the pydantic validation was not seeing the joined tables. I updated my code to add eager loading and it is now working exactly as I expect it to.

```
test = relationship("Test", lazy='joined')
```
